### PR TITLE
build: Select stage/prod env based on $(VERSION)

### PR DESCRIPTION
**Pull request by camscale**
Select the appropriate `stage` or `prod` GitHub Actions environment when
running `tag-build` or `tag-publish` based on `$(VERSION)`. If a version
string has no pre-release suffix, or it has `-cloud.` then it is considered a prod release and will be
built/published in the `build-prod`/`publish-prod` environment. Other
pre-release tags will be built/published in the
`build-stage`/`publish-stage` environment.

This can be overridden by specifying the environment name when building
or publishing:

    make ENVIRONMENT=build-prod tag-build
    make ENVIRONMENT=publish-prod tag-publish

### CODE REVIEW

File: Makefile
Comment (camscale) [inline Makefile:1230]: Since we're not ready for the tag-publish workflow to take `publish-stage` as an environment yet, perhaps a better approach to this is to just update the `gh` call to add `-f "environment=publish-prod"` (and `-f "environment=build-prod"` in `tag-build`) now and merge this PR after both the build and publish workflows work with stage?

Comment (fheinecke) [reply Makefile:1230]: How about this for now, and revert later?
```suggestion
tag-publish: ENVIRONMENT = publish-prod
# tag-publish: ENVIRONMENT = $(if $(IS_PROD_SEMVER),publish-prod,publish-stage)
```

Stage publishing just won't work for now. I'll be working on publishing immediately after the other work lands (and backports are filed), so I don't think it's a huge deal.

Comment (camscale) [reply Makefile:1230]: I'll put this PR on hold until the publishing can be done to stage. I think the two need to be consistent. Developers are building releases right now for which they want to access the build artifacts via publishing, so the two steps do need to be consistent.

**Review by fheinecke — APPROVED** [review]

**Comment by fheinecke** [comment]
This needs an e ref update and to be merged in sync with https://github.com/gravitational/teleport.e/pull/4122

**Review by r0mant — APPROVED** [review]

**Pull request merged**
